### DISCUSSION TITLE
Mixed requester station not working

### DISCUSSION THREAD

**Discussion by Wh1tePhant0m** [Opinionated-Blueprints/10-Books-Full-of-Rails#106] [Q&A]
Trying to pull 3 different items to one station for the mall. I have 6 stations each needs 3 unique items to be requested. I set the items to be requested in the combinator on the second row of signals.

![Screenshot 2024-12-05 134233](https://github.com/user-attachments/assets/00169da1-2696-4fd0-bd79-6f7d615cc658)
as an example here I want to pull plate, gears and sticks. But the the trains fill up at the stick station and then cant find where to go...

![Screenshot 2024-12-05 134333](https://github.com/user-attachments/assets/96e88108-b6a2-4dd4-9e31-930195886df3)

The train is searching the the station in the group with the [(Group)-(Requester)-(Item on train)] name
![Screenshot 2024-12-05 135224](https://github.com/user-attachments/assets/b382ee0e-e11f-4fd5-8519-385c361ef803)

But using the V2.22.x mixed Requester station the station name is set to the first item when inputting the parameter dialogue, However Station names are a plane string and the green item parameter signal doesn't work as a variable parameter in the station name.
![Screenshot 2024-12-05 135345](https://github.com/user-attachments/assets/ec48a949-b197-4ebe-a250-0f8e97ea5e62)

Am I just doing this all wrong or is there an issue here?

Both provider and requester are from the same V2.22.0 book, as far as I can see there are now track signal problems for the train pathing etc.

Thanks for the help in advance, this BP book is freaking amazing

**Comment by Wh1tePhant0m** [comment]
I Found a solution to multi item requests. The mighty   /   can be used in the station name as an or function.
![image](https://github.com/user-attachments/assets/b1ba61b5-0a98-4a93-a751-21541b2692be)
you can prob just set the BP parameters for the station name as such...
![image](https://github.com/user-attachments/assets/9baa3b11-6e01-4300-8fb9-617bd1ac835e)

**Reply by ghost** [reply]
> The mighty / can be used in the station name as an or function.

Could you elaborate? How does this work? What does the interrupt you're using look like?

**Comment by Wh1tePhant0m** [comment]
BP string for updated mixed requester:
0eNrtWutu2zgWfhVCfzrtyIl1tR0gC2yy2OkADTaYFAgWiWHQFm0TlUQNRcX1Fn6AfY99sn2SPYeSLNmWLSsZoFNgMZ2Yong+HpIfz4XUN2MaZiyRPFbG1TcjYOlM8kRxERtXxj9FRgIRv1MkZiwglKQsoZIqRhZSZAmZC0ki/hVeccWi9OI5RokVD0MyZSSRIkoUvEwFCIZspgiNdcsL8tcgIGrJiGQp1M4Vkysqg5RM1wQ0YZLOFI8XAKWWut1MxKmisYJCNOUxVUJCb58l5XGad5hQbCoKfST7PQNoJgGSpiwmItY42DtZg5IJn32BrnRlIpIsAbhf5/rVkr4wEmWh4knIQHOVEjGvjbMOjgoWKCmNinkxc1TJIyrXuUiUwTinOI4IQBUL1ySLOcCgympJFfaDZfoiOExNPrCFwEnQLRhZSQEPMAu4OBeGaXCcFOPq6ZuR8kVMQ1w/tU4YLNwLlyqDGtOIQSuoyFv0UJceLmEEOkhjAyBxwL4aV9bG7ABzV5O0N2PTYLHiirNcG/2wnsRZNIU+rqytNJUwW9Azn/WqZQTwRKQ8Z9w3AxEHgwvPNNbGVc/1B9BTwCWQRzcYmgaMWkkRTqYMFooDAEhVyBN4HWi0FF/MuUzV5OyBwezOvvRS/i+GI0wZgp0vfYtCIgH2FvvnEpqITCVZBxU+GZsNwCQhXTM52d2PT4VQL297faD1mFyTv0sGZAwF0BaYOc3mcyaf4yOityjxWSga5pwjM5rQGSwf+QkeNHD6/qj0J5S+peEsCyludL0fyScecUV+iuhXHmURyXmAW0j3EAoK+1xvmJzK0GVM5ly9N5CDe+Sxt9MTsBkPmGxhjl/yxr/wdplj2c3UKXD3eFN/eupMo0+vYY8WQvuAFhZ1M/777/9Aq3rfk5iplZBftI6SBcbVnIYpM40FLDoMUsmM7Xd9KIOttiIaYIObOGdqR4uitT5Hq64GZjM+vg3QjF1r8N4KVnh8ip5Aw4LaIfISuKylQc122VlF7R2If+iZggbnYoGrEgsGpJeVSxP6saAfKSafHFdnDu6UHAydgAve08IkqyWfLdkLwPKUhGLF5EXT5nK6W+bR97LMjM6Wr9lVpVzdLH94hVlGWm/aNmPDxmrdjE1b5Rjx7yUsaqxy85kzUrMiZDDfEBxQXO38FyzwUv9AXEagCwHxCINASJVBT2F936VNsVUTW9zubBl+L7Z8fg1Vdp1/nTC9VxCmjtbVoX+uvHLuv4+4ZR6jGTorKkC0uOaLdVSaA9XDhHyU50FWYPOdkAPZVeI1EcnbTpWmMKCKpIk9TsmdMz25ZDSY4H6YaNxyO+qHCXaSsKAzhVQOlk606d/BLOo6hwVqAqmBAKavS7jy+XyoewM5VezhrUvcRva9/njrFYpcpQceIVVj0iP7ba3x5X6VfVjlHFa5h1XeYZU/NnC9wmKRjKs+WDv9d6qTAwrlC3c07Dsj2/Lgx3J9x7edvusPB96mgUD+djYiFvAs6unkUoI5SkTImqjkVYYIik2Yg+7WbQvpnhtr/lHm7f7V8WXdqv38Cqt23+4E/6iItKPNvEeThLk+KXfT6bSlZr2yMIQ4LOQQMEEOC/YLoqZ6cpJI8QJB2nP8gGcBJTzm5Uc7+Pmolk0Wcdgxy3ErOvtn+dU/cZLzlx80xWnq4f8pzp8txRm96fDJ2/w4Sc3lG2PU723W/8bRxqYEx5ZHp9uwVJ8Kq3dpHq8S1BeN7wJPTIFle5Z8P6htooXV784L/8fkxWuS3RLn+2a72tvOeagP2mG981y2TGPBN09FYWdyy0XjoD2zfY7BUKGDh3+UpHTOwJNHjKaZZPo+Y7VkOoWBRwr/RwL+qCVEAR7mXimToE6hFdjKVGk6vi+Uqx9rZsVRZ1oc8JfXD5iOY6qyLlrgxQr0MYMkHqYot4HYVoNH9Asm9lJAJWqXi+hbgdK+po2Gz6qO3ctZOP9o5yCgPRJWpPn7dLcMvrW8VUAK6fXbq231uKbxO7zQKZIRCxnpRjtxw7WuyPDaSl9elNcRZnsI2wnc69fQnVb0x27odg3cbQV/6Abu1lX3zPbri07owzq634p+01F304CwZlJOk+UO3KHju4Nan4PWPj927HPY3unQbD846NSpM3Rr8COzPW7sBN+rb46aUail/6/fbJbdgGe/Ac9pwHPegOc24LlvwPMa8Lw34PkNeH5XvPGJPOJBu2Pym1hBKP9bcRYUkF/RD53Ojw+Tijfn3Y95wiEXgqzoAnzF0ZYPey3zw8Xn+PHDw/Wpm0y1c5OJ4tXBadtl5o0eSHHcimdlEGtgMoP9HxX6qBXF1pWKNx8+Xp862FXNB7tt+t1pWT009jUBb5uSu+uehYFuLdhsCgXszudlo9bzstpWzb/tEPL8y4Bj8XM9bt3NTRrJ/QsGiLWcADIACglAHmSdjv8rw5DilC6Wqgc/4cljaGc3JnKbcL3zcb0uuP75uIMuuIPzcUdHccemgVk5BnlPlmmZrumMTSjZ+F9RcouSA2ULSvDXHBYlG0q2LrlQcqHkmNjSKUqQrVm6CEKWbgnvTE9XuvB+pFu61WsPigMN5KN43qOP73OFfBS2TQ+Kg60eg60eg60eIxTPFRlV4qDPtisoO2UP+Bqb22OYka1Bz2PfwqvzoHLov+C3QvgdT7Bj//uasA3t0WgT61DAOilgHwrYJwWcQwHnpIB7KOCeFPAOBbyTAv6hgF8XKOhsloUrw6oQ7sCR1T5MOS3n9SvBG3Rq9+VVzUkxu5K61W7nUfs3qH2hktNpiG8eT2O4/UYQ8oB+ZQ8pbYGqDSJ3M7eVQ9Oou3g3LXDDZrwm1W6OqCZU+d2Z5Gl1erDto1dfMQNNCppxeKg+TzSNkE5ZiKejc3Ae1wGb0yxUvakIg/GTvmG6xtbjO/2pXhntSPIgQh6kT5e6yfjpEqXxUuoFNqc2YZ5vj9zRyHNH9tCHrGnzP+qtFfo=

**Comment by Wh1tePhant0m** [comment]
Now you simply ad up to 6 items when placing the blueprint, if you have less than 6 simply leave the remaining item parameters blank
![image](https://github.com/user-attachments/assets/2047a439-e233-43be-8fce-a7317a1ab379)

**Comment by Wh1tePhant0m** [comment]
NEVER MIND grrrr
so it turns out there is a string length limit on train station names, so where the above purple parameter icons give you a station name of

[item=logistic-robot] [item=requester-chest] - [item=iron-plate]/[item=iron-gear-wheel]/[item=iron-stick]/[item=parameter-4]/[item=parameter-5]/[item=parameter-6]

the green signal parameter icon has much longer string length so the station name caps out after three icons....

[item=logistic-robot][item=requester-chest] - [virtual-signal=signal-item-parameter]/[virtual-signal=signal-item-parameter]/[virtual-signal=signal-item-parameter]/[virtual-signal=signal-item-parameter~~~~~~LIMIT REACHED

SO... Im gonna go nuke my network and AGAIN and re build it with a mix station that has a max 3 item request and see if it fixes things.
